core/modules/qana/PostPlugin.cc

```diff
@@ -142,19 +149,22 @@ PostPlugin::applyPhysical(QueryPlugin::Plan& p,
             }
             // Patch MergeFixup.
             context.needsMerge = true;
-        } // if((_limit != -1) || _orderBy)
-        if(_orderBy) {
-            // Remove orderby from parallel
-            // (no need to sort until we have all the results)
-            SelectStmtPtrVector::iterator i, e;
-            for(i=p.stmtParallel.begin(), e=p.stmtParallel.end(); i != e; ++i) {
-                (**i).setOrderBy(std::shared_ptr<query::OrderByClause>());
+            LOGF(logger, LOG_LVL_DEBUG, "Add merge operation");
+
+            if(_orderBy) {
+                // Remove orderby from parallel
+                // (no need to sort until we have all the results)
+                SelectStmtPtrVector::iterator i, e;
+                for(i=p.stmtParallel.begin(), e=p.stmtParallel.end(); i != e; ++i) {
```

Comment:
I like previous grouping better, we had
    if((_limit != -1) || _orderBy) {
        // do some stuff
    }
    if (orderBy) {
        // do some other stuff
    }

and your code does the same
    if((_limit != -1) || _orderBy) {
        // do some stuff
        if (orderBy) {
            // do some other stuff
        }
    }
but it is harder to read. Please revert.